lib/ukfile/include/uk/file/iovutil.h

```diff
@@ -0,0 +1,165 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+/* Convenience operations for dealing with iovec */
+
+#ifndef __UKFILE_IOVUTIL_H__
+#define __UKFILE_IOVUTIL_H__
+
+#include <string.h>
+#include <sys/uio.h>
+
+#include <uk/essentials.h>
+
+/**
+ * Zero out at most `len` bytes in memory regions described by `iov[iovcnt]`,
+ * starting at `*curp` offset from the buffer at `iov[*iovip]`.
+ *
+ * If the total remaining space in `iov` is less than `len`, exit early.
+ * After the call, `*iovip` and `*curp` are updated with the new positions.
+ *
+ * @return Number of bytes zeroed
+ */
+static inline
+size_t uk_iov_zero(const struct iovec *iov, int iovcnt, size_t len,
+		   int *iovip, size_t *curp)
+{
+	size_t ret = 0;
+	int i = *iovip;
+	size_t cur = *curp;
+
+	if (cur) {
+		const size_t l = MIN(len, iov[i].iov_len - cur);
+
+		memset((char *)iov[i].iov_base + cur, 0, l);
+		ret += l;
+		len -= l;
+		cur += l;
+		if (cur == iov[i].iov_len) {
+			i++;
+			cur = 0;
+		}
+	}
+	if (len) {
+		UK_ASSERT(!cur);
```

Comment:
This check is not needed right? There is no way to reach this point with `cur != 0`, unless in the above `if` it ends up being larger than `iov[i].ion_len` which can happen only if it was originally larger that that (which should be checked initially as said above)

Reply:
In theory it is indeed not needed, as you said previous code ensures this precondition.
However, I generally find asserts useful not only for checking against undefined inputs, but also to express non-trivial preassumptions of a block of code, both for the computer and programmer.
Obviously simple cases like
```c
int x = 0;
UK_ASSERT(!x);
```
are redundant, as you can trivially tell at a glance that the condition is always true, but the more complex it gets, the more time and context you need to deduce these preassumptions. You could then add comments like `/* We know x to be 0 from now on */` but those are inferior to asserts IMO.

This specific case I agree is borderline trivial, but I consider the added clarity and programmer context this provides to be worth the small execution time cost in a debug build.
That said, I could be persuaded to remove the assert if you think the previous code is trivial enough.